rotary_utils/repair.py

```diff
@@ -472,59 +499,26 @@ def iterate_linking_contig_ends(contig_record: SeqIO.SeqRecord, bam_filepath: st
-                           f'overcome this, the script will shorter the circlator_min_length for this iteration to 90% '
-                           f'of the length threshold, i.e., {revised_min_length} bp.')
-
-            cli_tool_settings_dict_used = cli_tool_settings_dict_revised
-
-        else:
-            cli_tool_settings_dict_used = cli_tool_settings_dict
-
-        # Try to stitch the contig ends
         logger.info(f'Attempting reassembly with a length threshold of {length_threshold} bp')
-        length_outdir = os.path.join(linking_outdir, 'tmp', f'L{length_threshold}')
-
+        stitch_dirs.set_length_iteration_dirs(length_threshold, make_dirs=True)
+        override_circlator_min_length = override_min_stitch_length(tool_settings.circlator_min_length, length_threshold)
         flye_exit_status = link_contig_ends(contig_record=contig_record, bam_filepath=bam_filepath,
-                                            length_outdir=length_outdir, length_threshold=length_threshold,
-                                            cli_tool_settings_dict=cli_tool_settings_dict_used,
-                                            verbose_logfile=verbose_logfile, threads=threads)
-
-        # Make the output logging directory (this is needed even if Flye fails so the pipeline can keep going)
-        log_dir = os.path.join(log_dir_base, f'L{length_threshold}')
-        os.makedirs(log_dir, exist_ok=True)
-
-        # Stop early if Flye fails
+                                            length_outdir=stitch_dirs.length_outdir, length_threshold=length_threshold,
+                                            tool_settings=tool_settings, verbose_logfile=verbose_logfile,
+                                            override_circlator_min_length=override_circlator_min_length)
         if flye_exit_status != 0:
             assembly_attempts = assembly_attempts + 1
             continue
 
-        # If Flye was successful, copy important log files for that specific length threshold
-        shutil.copy(os.path.join(length_outdir, 'assembly', 'assembly_info.txt'), log_dir)
-        shutil.copy(os.path.join(length_outdir, 'merge', 'merge.circularise.log'), log_dir)
-        shutil.copy(os.path.join(length_outdir, 'merge', 'merge.circularise_details.log'), log_dir)
+        # Copy important log files
+        shutil.copy(os.path.join(stitch_dirs.length_outdir, 'assembly', 'assembly_info.txt'), stitch_dirs.log_dir)
+        shutil.copy(os.path.join(stitch_dirs.length_outdir, 'merge', 'merge.circularise.log'), stitch_dirs.log_dir)
+        shutil.copy(os.path.join(stitch_dirs.length_outdir, 'merge', 'merge.circularise_details.log'),
+                    stitch_dirs.log_dir)
 
-        # See if stitching the contigs ends worked
-        if check_circlator_success(os.path.join(length_outdir, 'merge', 'merge.circularise.log')) is True:
+        if check_circlator_success(os.path.join(stitch_dirs.length_outdir, 'merge', 'merge.circularise.log')) is True:
```

Comment:
@jmtsuji If the function returns True and False, then the `is True` is redundant and can be removed.

Reply:
Thanks -- `is True` is now removed.